src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderGetMethodExtension.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Doctrine\QueryItemExtension;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+final class OrderGetMethodExtension implements QueryItemExtensionInterface
+{
+    /** @var UserContextInterface */
+    private $userContext;
+
+    public function __construct(UserContextInterface $userContext)
+    {
+        $this->userContext = $userContext;
+    }
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return is_a($resourceClass, OrderInterface::class, true);
+    }
+
+    public function applyToItem(
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        array $identifiers,
+        string $operationName = null,
+        array $context = []
+    ) {
+        if (!is_a($resourceClass, OrderInterface::class, true)) {
+            return;
+        }
+
+        $rootAlias = $queryBuilder->getRootAliases()[0];
+        $user = $this->userContext->getUser();
+
+        $operationName = strtoupper($operationName);
+
+        $this->applyToItemForGetMethod($user, $queryBuilder, $operationName, $rootAlias);
+    }
+
+    private function applyToItemForGetMethod(
+        ?UserInterface $user,
+        QueryBuilder $queryBuilder,
+        string $operationName,
+        string $rootAlias
+    ): void {
+        if ($operationName === Request::METHOD_GET) {
```

Comment:
Move this `if` to `applyToItem` method next to https://github.com/Sylius/Sylius/pull/11548/files#diff-45ec3630ca31026139f7ff04770f08ccR50 to have all checks in one place
OR
move it to `supports` if it is used